configserver/src/main/java/com/yahoo/vespa/config/server/monitoring/ZKMetricUpdater.java

```diff
@@ -0,0 +1,127 @@
+package com.yahoo.vespa.config.server.monitoring;
+
+import com.yahoo.cloud.config.ZookeeperServerConfig;
+import com.yahoo.container.jdisc.config.HealthMonitorConfig;
+
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.net.InetAddress;
+import java.net.InetSocketAddress;
+import java.nio.ByteBuffer;
+import java.nio.channels.AsynchronousSocketChannel;
+import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Timer;
+import java.util.TimerTask;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static com.yahoo.vespa.config.server.monitoring.Metrics.getMetricName;
+
+public class ZKMetricUpdater extends TimerTask {
+    private static final Logger log = Logger.getLogger(ZKMetricUpdater.class.getName());
+
+    public static final String METRIC_ZK_ZNODES = getMetricName("zkZNodes");
+    public static final String METRIC_ZK_LATENCY_AVERAGE = getMetricName("zkAvgLatency");
+    public static final String METRIC_ZK_LATENCY_MAX = getMetricName("zkMaxLatency");
+    public static final String METRIC_ZK_CONNECTIONS = getMetricName("zkConnections");
+    public static final String METRIC_ZK_OUTSTANDING_CONNECTIONS = getMetricName("zkOutstandingConnections");
+
+    private final int CONNECTION_TIMEOUT_MS = 500;
+    private final int WRITE_TIMEOUT_MS = 250;
+    private final int READ_TIMEOUT_MS = 500;
+
+    private AtomicReference<Map<String, Long>> zkMetrics = new AtomicReference<>(new HashMap<>());
+    private final Timer timer = new Timer();
+    private final int zkPort;
+
+    public ZKMetricUpdater(Metrics metrics, ZookeeperServerConfig zkServerConfig, HealthMonitorConfig healthMonitorConfig) {
+        this(zkServerConfig, 4500, (long) (healthMonitorConfig.snapshot_interval() * 1000));
```

Comment:
metrics seems to be unused. Remove?

Reply:
Make 4500 a constant? Why 4500, btw? Make it dependent on healthMonitorConfig.snapshot_interval() instead?

Reply:
`metrics` is a leftover from how the code worked a bit earlier, will remove before merge. 4500 reflects a similar constant in `Metrics` where the initial delay is 5000. The idea is to try and trigger this one slightly before. Of course drift might make that a pointless exercise anyway.